compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -4374,6 +4377,64 @@ private STNode parseObjectTypeDescriptor() {
                 objectMembers, closeBrace);
     }
 
+    /**
+     * Parse object constructor expression.
+     *
+     * object-constructor-expr :=
+     *    [annots] [client] object [type-reference] {
+     *       object-member*
+     *    }
+     * object-member := object-field | method-defn
+     *
+     * @param isRhsExpr
+     * @param annots
+     * @return Parsed node
+     */
+    private STNode parseObjectConstructorExpressionNode(boolean isRhsExpr, STNode annots) {
+
+        startContext(ParserRuleContext.OBJECT_CONSTRUCTOR);
+
+        STNode objectTypeQualifier;
+        STToken nextToken = peek();
+        if (nextToken.kind == SyntaxKind.CLIENT_KEYWORD) {
+            startContext(ParserRuleContext.CLIENT_KEYWORD);
+            objectTypeQualifier = parseClientKeyword();
+            endContext();
+        } else {
+            objectTypeQualifier = STNodeFactory.createEmptyNode();
+        }
+
+        STNode objectKeyword = parseObjectKeyword();
+
+        STNode typeDescriptor;
+        nextToken = peek();
+        if (nextToken.kind == SyntaxKind.IDENTIFIER_TOKEN) {
+            typeDescriptor = parseTypeDescriptor(ParserRuleContext.TYPE_DESC_IN_NEW_EXPR);
```

Comment:
we need `type-reference := identifier | qualified-identifier` right? not a `typeDescriptor`
can use `parseQualifiedIdentifier()` for that.

Reply:
Please help check used `parseQualifiedIdentifier(ParserRuleContext.TYPE_REFERENCE);` in `parseObjectConstructorExpressionTypeReference()`

Reply:
Current way of parsing looks OK to me!